### PR TITLE
subhadip adhikary

### PR DESCRIPTION
**Pull request by subhadip007**
hi

### CODE REVIEW

File: Inc/stm32f446xx_gpio.h [hunk 1]
```diff
@@ -10,27 +10,51 @@
 
 #include<stdint.h>
 #include"stm32f446xx.h"
+#include <iostream>
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_gpio.h]: no need here. iostream works only in full fledged g++ environ. uC uses arm-g++ compiler, making iostream irrelevant.

**Review by CodeOn-ArK — COMMENTED** [review]
just started. Will be continuing soon

File: Inc/stm32f446xx_gpio.h (hunk 1 above)
Comment (subhadip007) [reply Inc/stm32f446xx_gpio.h]: ok

File: Inc/stm32f446xx_gpio.h [hunk 2]
```diff
@@ -11,26 +11,50 @@
...
+             __vo uint8_t GPIO_PinNumber;							// CAN BE ANYONE OF @GPIO_PIN_NUMBERS
+	         __vo uint8_t GPIO_PinMode;								// CAN BE ANYONE OF @GPIO_MODES
+	         __vo uint8_t GPIO_PinSpeed;								// CAN BE ANYONE OF @GPIO_SPEED
+	         __vo uint8_t GPIO_PinPuPdControl;						// CAN BE ANYONE OF @GPIO_PULL_UP_PULL_DOWN
+	         __vo uint8_t GPIO_PinOPType;							// CAN BE ANYONE OF @GPIO_OUTPUT_TYPE
+	         __vo uint8_t GPIO_PinAltFunMode;
+
+
+
+GPIO_PinConfig_t():  GPIO_PinNumber(0), GPIO_PinMode(0),GPIO_PinSpeed(GPIO_SPEED_HIGH),GPIO_PinPuPdControl(GPIO_NO_PUPD	), GPIO_PinOPType(GPIO_OP_TYPE_PP), GPIO_PinAltFunMode(0){};
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_gpio.h]: keep GPIO_PinSPeed = LOW

File: Inc/stm32f446xx_gpio.h [hunk 3]
```diff
@@ -11,26 +11,50 @@
...
+	         __vo uint8_t GPIO_PinPuPdControl;						// CAN BE ANYONE OF @GPIO_PULL_UP_PULL_DOWN
+	         __vo uint8_t GPIO_PinOPType;							// CAN BE ANYONE OF @GPIO_OUTPUT_TYPE
+	         __vo uint8_t GPIO_PinAltFunMode;
+
+
+
+GPIO_PinConfig_t():  GPIO_PinNumber(0), GPIO_PinMode(0),GPIO_PinSpeed(GPIO_SPEED_HIGH),GPIO_PinPuPdControl(GPIO_NO_PUPD	), GPIO_PinOPType(GPIO_OP_TYPE_PP), GPIO_PinAltFunMode(0){}; 
+
+protected:
+void set_Pin(uint8_t pin=0,uint8_t pinmode=0, uint8_t pinaltfunmode=0) {
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_gpio.h:30]: also set other fields like OPType, PuPdControl, PinSpeed

File: Inc/stm32f446xx_i2c.h [hunk 4]
```diff
@@ -11,34 +11,46 @@
...
  * HANDLE		STRUCTURE		FOR 		I2C 		PERIPHERAL
  */
-typedef struct{
+class I2C_Handle_t:public I2C_Config_t
+{
 
-	I2C_RegDef_t	*pI2Cx;
-	I2C_Config_t	I2C_Config;
 
-	uint8_t			*pTxBuffer;			// !<	  To store the app. Tx buffer address 	>
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_i2c.h:32]: Add these also

File: Inc/stm32f446xx_i2c.h [hunk 5]
```diff
@@ -70,17 +82,6 @@ typedef struct{
 #define  I2C_FM_DUTY_2				0
 #define  I2C_FM_DUTY_16_9			1
 
-/*
- * @I2C_application_state
- */
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_i2c.h]: Add the I2C app states also

File: Inc/stm32f446xx_i2c.h [hunk 6]
```diff
@@ -108,15 +109,8 @@ typedef struct{
 #define I2C_FLAG_SR2_DUALF		7
 #define I2C_FLAG_SR2_PEC			8
 
-/*
- * I2C EVENT MACROS
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_i2c.h]: Add I2C event MAcros

File: Inc/stm32f446xx_i2c.h [hunk 7]
```diff
@@ -136,22 +130,12 @@ void I2C_Init(I2C_Handle_t *pI2CHandle);									//INITIALIZES I2C PORT
...
 
 /*
- *BLOCKING		 DATA TRANSMIT / RECIEVE
+ * DATA TRANSMIT / RECIEVE
  */
 
 void I2C_MasterSendData(I2C_Handle_t *pI2CHandle, uint8_t *pTXBuffer, uint32_t Len, uint8_t SlaveAddr);
 void I2C_MasterRecieveData(I2C_Handle_t *pI2CHandle, uint8_t *pRXBuffer, uint32_t Len, uint8_t SlaveAddr);
 
-
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_i2c.h]: Add interrupt data transmit ? receive also

File: Inc/stm32f446xx_i2c.h [hunk 8]
```diff
@@ -160,9 +144,6 @@ void I2C_CloseRecieveData(I2C_Handle_t *pI2CHandle);
 void I2C_IRQ_ITConfig(uint8_t IRQNumber, uint8_t En_Di);					//CONFIGURES IRQ
 void I2C_IRQConfig(uint8_t IRQNumber,uint32_t IRQPriority);					// I2C PRIORITY HANDLER
 
-void I2C_EV_IRQHandling(I2C_Handle_t *pI2CHandle);
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_i2c.h]: Add these

File: Inc/stm32f446xx_spi.h [hunk 9]
```diff
@@ -23,22 +27,43 @@ typedef struct{
...
+class SPI_Handle_t: public SPI_Config_t
+{
 	/*
 	 * HANDLE STRUCT FOR SPIx PERIPHERAL
 	 */
+
+public:
 	SPI_RegDef_t *pSPIx;						//HOLDS THE ADDRESS OF THE SPIx(x : 1,2,3) PERIPHERAL ADDRESS
-	SPI_Config_t SPIConfig;
+	void SPI_Handle(uint8_t S1,uint8_t S2,uint8_t S3,uint8_t S4,uint8_t S5,uint8_t S6,uint8_t S7){
```
Comment (CodeOn-ArK) [inline Inc/stm32f446xx_spi.h:56]: Please Give full Names

**Review by CodeOn-ArK — CHANGES_REQUESTED** [review]
Please make the necessary changes. @subhadip007 da

File: Inc/stm32f446xx_gpio.h (hunk 3 above)
Comment (CodeOn-ArK) [reply Inc/stm32f446xx_gpio.h:30]: OPType = GPIO_OP_TYPE_PP; PuPdControl = GPIO_NO_PUPD; PinAltfun = GPIO_ALTFN_OP

**Review by CodeOn-ArK — CHANGES_REQUESTED** [review]

File: Inc/stm32f446xx_gpio.h (hunk 2 above)
Comment (subhadip007) [reply Inc/stm32f446xx_gpio.h]: passing macros here is fine r8 or i have to pass the actual values?

File: Inc/stm32f446xx_gpio.h (hunk 3 above)
Comment (subhadip007) [reply Inc/stm32f446xx_gpio.h:30]: PuPd and OPType are already initialized i am setting  the pinaltfun one to GPIO_ALTFN_OP

File: Inc/stm32f446xx_i2c.h (hunk 4 above)
Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h:32]: till
uint8_t			Sr	;						// !<	  To store Repeated start value	>

right?

File: Inc/stm32f446xx_i2c.h (hunk 5 above)
Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h]: done

File: Inc/stm32f446xx_i2c.h (hunk 6 above)
Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h]: done

File: Inc/stm32f446xx_i2c.h (hunk 7 above)
Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h]: done ,idk what happned

Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h]: i had to change  the whole i2c after that I2C_Handle_t basically

File: Inc/stm32f446xx_i2c.h (hunk 8 above)
Comment (subhadip007) [reply Inc/stm32f446xx_i2c.h]: done

File: Inc/stm32f446xx_spi.h (hunk 9 above)
Comment (subhadip007) [reply Inc/stm32f446xx_spi.h:56]: what do you mean?

**Review by CodeOn-ArK — APPROVED** [review]
file approved!!

File: Inc/stm32f446xx_spi.h (hunk 9 above)
Comment (CodeOn-ArK) [reply Inc/stm32f446xx_spi.h:56]: full names for S1, S2 etc

**Pull request merged**